From f6791d3baf2c07e1e78c6bb4750f24e616902537 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Tue, 29 Oct 2024 07:33:21 +0200
Subject: [PATCH] [SW-205076] gaudi2: security: open pcie elbi rr region

The elbi rr if special glbl spare register 2 and 3 are used
for arm tx channel in hl-smi. This patch adds support to open
elbi rr region using range register and other registers in this
region are protected via protection bits.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I30c3d445befdb287abd202fc69e645ce45fc3c80
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 02fd130..6fc87ea 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -164,7 +164,6 @@
 	HL_SEC_LBW_PCIE_PMA_01,
 	HL_SEC_LBW_PCIE_PMA_23,
 	HL_SEC_LBW_ELBI_FIT_REGION, /* unused in EROM. used in preboot. Hence Keeping it here */
-	HL_SEC_LBW_CPU_CA53_CFG,
 	HL_SEC_LBW_CPU_TIMESTAMP_MSTR_IF,
 	HL_SEC_LBW_PSOC_ARC_DBG,
 	HL_SEC_LBW_PSOC_CA53_DBG,
@@ -655,6 +654,7 @@
 /* protection bits configurations to open some selective registers */
 static u32 gaudi2_protbits_data0[] = { 0x400, 0xc103 };
 static u32 gaudi2_protbits_data1[] = { 0x3f, 0x40000000 };
+static u32 gaudi2_protbits_data3[] = { 0x0c000000 };
 
 struct hl_prot_bits_cfg gaudi2_prot_bits_cfg[] = {
 	{{ HL_BLOCK_TYPE_PSOC, (mmPSOC_GLOBAL_CONF_BASE & ~CFG_BASE), 1, 0, 0, 0x0, 0x0, 0x0 },
@@ -669,6 +669,13 @@
 	  .data = gaudi2_protbits_data1,
 	  .data_size = ARRAY_SIZE(gaudi2_protbits_data1)
 	},
+
+	{{ HL_BLOCK_TYPE_PCIE, 0xfcc14000, 1, 0, 0, 0x0, 0x0, 0x0 },
+	  .prot_map = 0x00000000,
+	  .data_map = 0x40000000,
+	  .data = gaudi2_protbits_data3,
+	  .data_size = ARRAY_SIZE(gaudi2_protbits_data3)
+	}
 };
 
 static struct hl_prot_bits_cfg gaudi2_prot_bits_reset[] = {
@@ -859,10 +866,18 @@
 	hl_secure_lbw_rr(HL_SEC_LBW_PCIE_WRAP, min, max, rr_sel);
 
 	/*
-	 * PCIE DBI base to PSOC GLOBAL CONF base
-	 * 0x1000007FFCC02000 - 0x1000007FFCC4B000
+	 * PCIE DBI base to PCIE ELBI MSTR IF base
+	 * 0x1000007FFCC02000 - 0x1000007FFCC14000
 	 */
 	min = mmPCIE_DBI_BASE;
+	max = mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_HBW_BASE;
+	hl_secure_lbw_rr(HL_SEC_LBW_CPU_CA53_CFG, min, max, rr_sel);
+
+	/*
+	 * PCIE MSTR RR IF to PSOC GLOBAL CONF base
+	 * 0x1000007FFCC15000 - 0x1000007FFCC4B000
+	 */
+	min = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_BASE;
 	max = mmPSOC_GLOBAL_CONF_BASE;
 	hl_secure_lbw_rr(HL_SEC_LBW_DBG_CFG_SPACE, min, max, rr_sel);
 }
